include/aca_comm_mgr.h

```diff
@@ -24,66 +25,24 @@ namespace aca_comm_manager
 {
 class Aca_Comm_Manager {
   public:
-  // constructor and destructor purposely omitted to use the default one
-  // provided by the compiler
-  // Aca_Comm_Manager();
-  // ~Aca_Comm_Manager();
-
   static Aca_Comm_Manager &get_instance();
 
   int deserialize(const cppkafka::Buffer *kafka_buffer,
                   alcorcontroller::GoalState &parsed_struct);
 
-  // process ONE VPC state
-  int update_vpc_state_workitem(const alcorcontroller::VpcState current_VpcState,
-                                alcorcontroller::GoalStateOperationReply &gsOperationReply);
-
-  // process 0 to N VPC states
-  int update_vpc_states(const alcorcontroller::GoalState &parsed_struct,
-                        alcorcontroller::GoalStateOperationReply &gsOperationReply);
-
-  // process ONE subnet state
-  int update_subnet_state_workitem(const alcorcontroller::SubnetState current_SubnetState,
-                                   alcorcontroller::GoalStateOperationReply &gsOperationReply);
-
-  // process 0 to N subnet states
-  int update_subnet_states(const alcorcontroller::GoalState &parsed_struct,
-                           alcorcontroller::GoalStateOperationReply &gsOperationReply);
-
-  // process ONE port state
-  int update_port_state_workitem(const alcorcontroller::PortState current_PortState,
-                                 const alcorcontroller::GoalState &parsed_struct,
-                                 alcorcontroller::GoalStateOperationReply &gsOperationReply);
-
-  // process 0 to N port states
-  int update_port_states(const alcorcontroller::GoalState &parsed_struct,
-                         alcorcontroller::GoalStateOperationReply &gsOperationReply);
-
-  int update_goal_state(const alcorcontroller::GoalState &parsed_struct,
+  int update_goal_state(alcorcontroller::GoalState &parsed_struct,
```

Comment:
So the only API exposed by ACA is update_goal_state?

Reply:
the communication manager also expose deserialize() for kafka message handing.

Reply:
Okay I guess my actual question is "What is the boundary line between comm_manager and netw state handler?"  I see some codes are moving to the handler class from comm_manager.

Reply:
Sure, the components boundary is modeled based on the design diagram: https://github.com/futurewei-cloud/alcor-control-agent/blob/master/docs/images/Agent_components.jpg